src/plugins/intel_cpu/src/nodes/eltwise.cpp

```diff
@@ -157,88 +160,175 @@ class EltwiseShapeInferFactory : public ShapeInferFactory {
     }
 };
 
+void set_intersection(const std::set<std::vector<element::Type>>& precisions1,
+                      const std::set<std::vector<element::Type>>& precisions2,
+                      std::set<std::vector<element::Type>>& intersection) {
+                      std::map<element::Type, size_t> intersection_types;
+
+    for (auto it1 = precisions1.begin(); it1 != precisions1.end(); ++it1) {
+        for (auto it2 = precisions2.begin(); it2 != precisions2.end(); ++it2) {
+            const auto& it1_precisions = *it1;
+            // all element types are equal
```

Comment:
How this is guaranteed? Is it just an assumtion that all eltwise emitters work only with equal precisions on inputs? If so I would suggest to add aditional check that only sets with equal precisions should be used in intersection computation